js/src/builder/struct.ts

```diff
@@ -23,12 +23,31 @@ import { Struct, TypeMap } from '../type.js';
 
 /** @ignore */
 export class StructBuilder<T extends TypeMap = any, TNull = any> extends Builder<Struct<T>, TNull> {
-    public setValue(index: number, value: Struct<T>['TValue']) {
+    /**
+     * Write a value (or null-value sentinel) at the supplied index.
+     * If the value matches one of the null-value representations, a 1-bit is
+     * written to the null `BitmapBufferBuilder`, otherwise, a 0 is written. The
+     * value is then passed to `Builder.prototype.setValue()`.
+     * @param {number} index The index of the value to write.
+     * @param {T['TValue'] | TNull } value The value to write at the supplied index.
+     * @returns {this} The updated `Builder` instance.
+     */
+    public set(index: number, value: Struct<T>['TValue'] | TNull) {
+        this.setValid(index, this.isValid(value));
+        this.setValue(index, value);
+        return this;
+    }
```

Comment:
```suggestion
    /** @inheritdoc */
    public set(index: number, value: Struct<T>['TValue'] | TNull) {
        if (this.setValid(index, this.isValid(value))) {
            this.setValue(index, value);
        } else {
          this.children.forEach((child) => child.setValid(index, false));
        }
        return this;
    }
```

Reply:
Is `inheritdoc` correct here? I believe the inherited documentation would be misleading as it wouldn't mention calling the `setValid` function on the children.

Also I'll admit I'm very unfamiliar with this codebase, is passing `false` good enough, I would have thought you need to pass `value`?

Reply:
Is `inheritdoc` correct here? I believe the inherited documentation would be misleading as it wouldn't mention calling the `setValid` function on the children.

Also I'll admit I'm very unfamiliar with this codebase, is calling `setValid` good enough, I would have thought that doesn't work for more complex nested types such as a struct within a struct which would need the fields _of that inner-struct_ to also have a null value appended.

Reply:
> I would have thought that doesn't work for more complex nested types such as a struct within a struct which would need the fields of that inner-struct to also have a null value appended.

@Alfred-Mountfield Ah good point. In that case, I think the change needs to be not to override `set()` or change the `setValue()` implementation, but to also override `setValid()` and make it set nulls into each child. I'll update my suggestion in a new thread.

Reply:
I've just tested it locally and that seems to work :) although with `valid` instead of value` in `if (!super.setValid(index, value)`

There doesn't seem to be documentation on the `setValid` method on `Builder` however, so I'm not sure what doc it's inheriting